.build/images/dietpi-imager

```diff
@@ -338,6 +338,15 @@
 		# Remount image for any required edits
 		G_EXEC mkdir "$FP_MNT_TMP"
 		G_EXEC mount "$FP_ROOT_DEV" "$FP_MNT_TMP"
+		[[ -z ${ADD_DOS_PART+x} ]] && ADD_DOS_PART="0"
+		if [[ "$ADD_DOS_PART" == "1" ]]
+		then
+			# let's copy the two config files we want to add to the setup partition
+			TMP_DIETPI_CONF_DIR=$(mktemp -d)
+			cp "$FP_MNT_TMP"/boot/dietpi.txt "$TMP_DIETPI_CONF_DIR"/
+			cp "$FP_MNT_TMP"/boot/dietpi-wifi.txt "$TMP_DIETPI_CONF_DIR"/
+			G_EXEC_OUTPUT=1 G_EXEC ls -l "$TMP_DIETPI_CONF_DIR"/
+		fi
```

Comment:
I suggest to just mount the rootfs and the new partition both in the same block below. I see no reason to copy the config files separately into a tmp dir here, and then from there to the new fs below. The rootfs could be mounted R/O down there to rule out any accidental writes.

Also the still existing loop device could be used instead of removing it first and creating a new one.
```sh
losetup -c "$FP_SOURCE"
```
can be used to re-read the image size. Also note that at this point it is only needed if `[[ $FP_SOURCE_IMG ]]`, else the source is a drive, `dd` can/must be skipped and the new partition mounted directly after `partprobe`/`partx -u`.

Reply:
I'll need to look into that comment when I had more coffee (it's only 7am here). I think I know what you are telling me. Simplify things by mounting the rootfs r/o, reusing the existing loop device.
And pay attention to the case where source is a drive.

Reply:
Right, and having all code related to this feature merged into a single code block 🙂. Enjoy your coffee and breakfast.